app/models/maintenance_tasks/batch_csv_collection_builder.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "csv"
+
+module MaintenanceTasks
+  # Strategy for building a Task that processes CSV files in batches.
+  #
+  # @api private
+  class BatchCsvCollectionBuilder
```

Comment:
You _could_ inherit `CsvCollectionBuilder`, to repeat less code here.

Reply:
Good point, thanks Gannon!